src/overlays/actors/ovl_Obj_Wturn/z_obj_wturn.c

```diff
@@ -13,42 +13,117 @@
 void ObjWturn_Init(Actor* thisx, GlobalContext* globalCtx);
 void ObjWturn_Update(Actor* thisx, GlobalContext* globalCtx);
 
+void func_808A7954(ObjWturn* this);
 void func_808A7968(ObjWturn* this, GlobalContext* globalCtx);
+void func_808A7A24(ObjWturn* this);
 void func_808A7A5C(ObjWturn* this, GlobalContext* globalCtx);
+void func_808A7AAC(ObjWturn* this, GlobalContext* globalCtx);
 void func_808A7BA0(ObjWturn* this, GlobalContext* globalCtx);
+void func_808A7C04(ObjWturn* this, GlobalContext* globalCtx);
 void func_808A7C78(ObjWturn* this, GlobalContext* globalCtx);
 
-#if 0
 const ActorInit Obj_Wturn_InitVars = {
-    ACTOR_OBJ_WTURN,
-    ACTORCAT_ITEMACTION,
-    FLAGS,
-    GAMEPLAY_KEEP,
-    sizeof(ObjWturn),
-    (ActorFunc)ObjWturn_Init,
-    (ActorFunc)Actor_Noop,
-    (ActorFunc)ObjWturn_Update,
-    (ActorFunc)NULL,
+    ACTOR_OBJ_WTURN,       ACTORCAT_ITEMACTION,        FLAGS,
+    GAMEPLAY_KEEP,         sizeof(ObjWturn),           (ActorFunc)ObjWturn_Init,
+    (ActorFunc)Actor_Noop, (ActorFunc)ObjWturn_Update, (ActorFunc)NULL,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/ObjWturn_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7954.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7968.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7A24.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7A5C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7AAC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7BA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7C04.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7C78.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/ObjWturn_Update.s")
+static Vec3f D_808A7DC0 = { 0.0f, -1.0f, 0.0f };
+
+void ObjWturn_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjWturn* this = THIS;
+
+    func_808A7954(this);
+}
+
+void func_808A7954(ObjWturn* this) {
+    this->actionFunc = func_808A7968;
+}
+
+void func_808A7968(ObjWturn* this, GlobalContext* globalCtx) {
+    if (globalCtx->msgCtx.unk1202A >= 28 && globalCtx->msgCtx.unk1202A < 39) {
+        Actor_UnsetSwitchFlag(globalCtx, this->actor.params);
+        Actor_MarkForDeath(&this->actor);
+    } else if ((Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == 88)) ||
+               (!Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == 89))) {
```

Comment:
`sceneNum` has an enum, these two are `SCENE_F40` and `SCENE_F41`, and the one later on is also `SCENE_F40`
```suggestion
    } else if ((Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == SCENE_F40)) ||
               (!Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == SCENE_F41))) {
```